### PR TITLE
Feat/map

### PR DESCRIPTION
**Pull request by D3nnis249**
Hello and thank you for your review. First of all, we would like to apologize for the quantity of this PR. 

First we added Mapbox, and later we decided to use react-leaflet. We added a map and mapped markers for each gas station. Then we added a location query and want to finish the map in the next steps. We decided to make a pull request, because otherwise it would take too long to check it.

### CODE REVIEW

File: client/src/components/Map.jsx [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+  return (
+    <MapWrapper center={[52.500478, 13.376696]} zoom={13} scrollWheelZoom={true}>
+      <TileLayer
+        url={
+          'https://api.mapbox.com/styles/v1/alexf090/cl2uhl8gx008p14qop192uczn/tiles/256/{z}/{x}/{y}@2x?access_token=' +
+          mapToken
+        }
+      />
+      {stations
+        .filter(station => station.fuelPrices[fuelValue] !== null)
```
Comment (julialoeschel) [inline client/src/components/Map.jsx]: This is fine, but if you want to make it even more readable, you could create a variable with your filtered results first :octocat:

File: client/src/components/Map.jsx [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+  top: 6.25rem;
+  right: 0;
+  left: 0;
+  bottom: 6rem;
+  overflow: hidden;
+  z-index: 10;
+`;
+
+const NewPopup = styled(Popup)`
+  /* width: 90px; */
```
Comment (julialoeschel) [inline client/src/components/Map.jsx]: please delete code that you dont need :)

File: client/src/components/OpeningTimes.jsx [hunk 3]
```diff
@@ -15,43 +15,50 @@ const OpeningTimes = ({ currentStation }) => {
...
-            {openingDay.monday[0].opening} - {openingDay.monday[0].closing}
+            {' '}
+            {openingDay.monday[0]?.opening} - {openingDay.monday[0]?.closing}
           </time>
         </Day>
         <Day>
           Dienstag:
           <time>
-            {openingDay.tuesday[0].opening} - {openingDay.tuesday[0].closing}
+            {' '}
```
Comment (julialoeschel) [inline client/src/components/OpeningTimes.jsx:25]:
```suggestion
    
```

Comment (julialoeschel) [reply client/src/components/OpeningTimes.jsx:25]: same for the next rows 🐜

File: client/src/pages/MapPage.jsx [hunk 4]
```diff
@@ -1,10 +1,33 @@
...
+    const map = useMapEvents({
+      click() {
+        map.locate();
+      },
+      locationfound(e) {
+        setPosition(e.latlng);
+        map.flyTo(e.latlng, map.getZoom());
+      },
+    });
+    return position === null ? null : (
```
Comment (julialoeschel) [inline client/src/pages/MapPage.jsx]: I guess this works, but maybe there is an easyer way ? 
`position? do smth : null; `

File: client/src/pages/MapPage.jsx [hunk 5]
```diff
@@ -1,10 +1,33 @@
...
 
 function MapPage({ gasInfoHead, title, fuelValue }) {
+  function LocationMarker() {
+    const [position, setPosition] = useState(null);
+
+    const map = useMapEvents({
+      click() {
+        map.locate();
+      },
+      locationfound(e) {
```
Comment (julialoeschel) [inline client/src/pages/MapPage.jsx]: its better readable if you write it out here.

**Review by julialoeschel — COMMENTED** [review]
Great job, 
I have a view suggestions for you. 
🐳

File: client/src/components/OpeningTimes.jsx (hunk 3 above)
Comment (D3nnis249) [reply client/src/components/OpeningTimes.jsx:25]: This is not a mistake even if it seems like one. We have inserted this to always have a blank space there ✌️😊

Comment (AlexF090) [reply client/src/components/OpeningTimes.jsx:25]: Example:
Without: "Dienstag:00:-00 - 00:00"
With: "Dienstag: 00:00 - 00:00"

File: client/src/pages/MapPage.jsx (hunk 5 above)
Comment (AlexF090) [reply client/src/pages/MapPage.jsx]: If you mean the "e", we've now rewritten it as "event". You can see it in the next commit. 😊

**Review by julialoeschel — APPROVED** [review]
👍

**Pull request merged**